### ISSUE TITLE
Allow tagging slack groups such as @custom_group to include the users

### ISSUE THREAD

**Issue by Allan-W-Smith** [dgurkaynak/slack-poker-planner#68]
Is this a feature people would be interested in?  I'm not sure how complicated this would be but if people are interested I'm happy to take a look.

**Comment by dgurkaynak** [comment]
Hey @Allan-W-Smith, sorry for the late response. 
My initial impression is this will be a bit complicated, the things that come to my mind:
- We're using Slack's built-in [user list multi-select input](https://api.slack.com/reference/block-kit/block-elements#users_multi_select) for participants selection. Now slack is handling everything: getting all the users in the workspace, listing them with their online/offline status, autocomplete etc...
- Unfortunately, it doesn't support user groups.
- If we're going to implement this, we probably will use [multi-select with external data source](https://api.slack.com/reference/block-kit/block-elements#external_multi_select). But now we're responsible for handling everything: getting all the users, getting all the usergroups...
- For now, the app requires just 2 slack permissions, which I think they're the bare minimum:
  - `commands` for adding `/pp` slash command
  - `chat:write` for sending messages to channels & conversations
- If we're gonna handle user & usergroups, I guess we need 2 more permissions: [`users:read`](https://api.slack.com/scopes/users:read) and [`usergroups:read`](https://api.slack.com/scopes/usergroups:read)
- To get these additional permissions, we have to redirect users to re-add poker planner app to their workspace again, because the current auth tokens don't have these extra permissions
- I'm kind of hesitant to request additional permissions, because I get lots of e-mail about what our app can and cannot access in their slack workspace. People have strong privacy concerns. That's why I put [a dedicated "Privacy" section](https://deniz.co/slack-poker-planner/#privacy) to app's home page 😅

**Comment by Allan-W-Smith** [comment]
Thanks for the response. Your considerations are very reasonable. 

Happy to contribute but I don't want to deter users from the project on privacy concerns.

My use case is based around devs being grouped by specialty. I don't need/want to tag a FE specialist into a BE ticket for PP. This is where I'm finding myself removing and adding users regularly depending on the ticket.

What about channel tagging? I assume we would have to leverage similar permissions?

**Comment by Allan-W-Smith** [comment]
Happy to close this if this isnt a direction you wisj to explore.

**Comment by dgurkaynak** [comment]
> What about channel tagging? I assume we would have to leverage similar permissions?

To get all the users in a channel or conversation, I guess we will use [`conversations.members`](https://api.slack.com/methods/conversations.members) API, and it requires additional permission scopes as well 😕

> My use case is based around devs being grouped by specialty. I don't need/want to tag a FE specialist into a BE ticket for PP. This is where I'm finding myself removing and adding users regularly depending on the ticket.

Currently, the app remembers the last settings you used to start a voting session per channel/conversation. What I mean by "settings" is the points and participants (basically everything you see in the session creation form, except for title). This is not an optional solution, but you can create 2 separate slack channels for FE and BE tickets. This can reduce the time you spend for changing participants for different types of tickets. FE tickets go to the FE channel, BE tickets go to the BE channel...

**Comment by dgurkaynak** [comment]
Maybe we can implement something like "settings profiles". Instead of remembering just the last used settings, user can save the current settings with a custom profile name, and then they can reuse one of those settings profiles?

**Comment by Allan-W-Smith** [comment]
Yep understood. I'm across current function and you make a good point on leveraging existing channels as a solution. It doesn't quite fit my needs as I have channels per pod + a dedicated estimations channel aswell as stack focussed channels. Was keen on fitting it into my existing channel structures. With 40+ channels running most of the time i definitely don't need additional channels. 

Happy for you to close this thread if we don't want to add additional permissions. 👍

**Comment by Allan-W-Smith** [comment]
Thinking default behaviour in a drop down with the ability to store and select additional profiles?

**Comment by dgurkaynak** [comment]
> Settings profiles could be a good solution if you would like me to explore that?

You're more than welcome if you want to 😊

- I think [we can get a callback](https://api.slack.com/reference/interaction-payloads/block-actions#payload_timing) when user changes a selectbox (before submitting).
- [We can update](https://api.slack.com/surfaces/modals/using#modifying) the layout of the session creation form modal.
- We can also push another modal views as well (like multi-stage modals), but I guess this will be more difficult to implement because we need to maintain state across those views.
- I guess we cannot change/modify the current state of the form inputs, but I'm not 100% sure.

Depending on my quick and dirty research about what we can and can't do, let me braindump further:
- We can put a settings profile dropdown under the title textbox:
![image](https://user-images.githubusercontent.com/1475786/191051184-796d0a0a-b0f9-4f02-9f08-5d0d5aee2695.png)
- In the dropdown, there should be always "Custom" option at the bottom. 
- If "Custom" option is chosen, we show all the form inputs that we currently have. Additionally, we need a text input for saving the current settings as a new profile.
- But if a real profile is chosen, we don't show the additional form inputs. Just "title" textbox and "settings profile" dropdown will be visible.
- If we can update the current state of the inputs, we can skip showing/hiding inputs. I think this will be more straightforward to implement, but it needs further investigation.

What do you think?  What is on your mind about the UI/UX?

**Comment by Allan-W-Smith** [comment]
I think that all sounds pretty reasonable to me.  I'll do some hands on investigation when I can find some time and report back.

Talk soon

**Comment by arvenil** [comment]
I think StoryPlan did it right. You can add it to channel and everyone in that channel can vote. Works perfectly - sadly other things aren't so I would prefer using Poker Planner but without the chore of picking ppl every single time :)

**Comment by dgurkaynak** [comment]
> I think StoryPlan did it right. You can add it to channel and everyone in that channel can vote. Works perfectly - sadly other things aren't so I would prefer using Poker Planner but without the chore of picking ppl every single time :)

@arvenil you shouldn't be manually choosing the participants every time, the last entered participants should be remembered per channel. Isn't that the case? What is your use case?
